File: app/src/main/java/com/example/myapplication/data/mapper/UserMapper.java
package com.example.myapplication.data.mapper;

import com.example.myapplication.data.remote.aidl.UserParcelable;
import com.example.myapplication.domain.model.User;

public final class UserMapper {

    private UserMapper() {
    }

    public static User toDomain(UserParcelable userParcelable) {
        return new User(
                userParcelable.getId(),
                userParcelable.getName(),
                userParcelable.getAge(),
                userParcelable.getWeight()
        );
    }

    public static UserParcelable toParcelable(User user) {
        return new UserParcelable(
                user.getId(),
                user.getName(),
                user.getAge(),
                user.getWeight()
        );
    }
}

File: app/src/main/java/com/example/myapplication/data/mapper/UserMapper.kt
package com.example.myapplication.data.mapper

import com.example.myapplication.data.remote.aidl.UserParcelable
import com.example.myapplication.domain.model.User

fun UserParcelable.toDomain(): User = User(
    id = id,
    name = name,
    age = age,
    weight = weight,
)

fun User.toParcelable(): UserParcelable = UserParcelable(
    id = id,
    name = name,
    age = age,
    weight = weight,
)
